x-pack/plugins/observability_solution/inventory/public/hooks/use_inventory_abortable_async.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useAbortableAsync } from '@kbn/observability-utils/hooks/use_abortable_async';
+import { i18n } from '@kbn/i18n';
+import { useKibana } from './use_kibana';
+
+export function useInventoryAbortableAsync<T>(...args: Parameters<typeof useAbortableAsync<T>>) {
+  const {
+    core: { notifications },
+  } = useKibana();
+  const response = useAbortableAsync(...args);
+
+  if (response.error) {
+    notifications.toasts.addDanger({
+      title: i18n.translate('xpack.inventory.apiCall.error.title', {
+        defaultMessage: `Error while fetching resource`,
+      }),
+      text: response.error.message,
```

Comment:
Shouldn't we also consider the `response.body?.message`? It would provide more details about the error.

Reply:
https://github.com/elastic/kibana/pull/193560/commits/31bf014c21dd8aab53305fef91e97b4c8ec12f4e